README.md

```diff
@@ -0,0 +1,335 @@
+
+```plaintext
+Alexa, ask Recipe Manager what do I need to cook Chocolate Chip Cookies
+```
+
+Having one intent in your Alexa Skill is no fun, so let's create another intent with its own set of sample phrases. Choose to create a new intent titled `GetRecipeFromIngredientsIntent` with the following sample utterances:
+
+```plaintext
+what can i cook with {ingredientone} and {ingredienttwo}
+what are some recipes with {ingredientone} and {ingredienttwo}
+if i have {ingredientone} and {ingredienttwo} what can i cook
+```
+
+This time around we're using two slot variables instead of one. Like previously mentioned, it is probably a good idea to add significantly more sample utterances to get the best results. Alexa needs to be able to process the data to send to your Lambda function.
+
+At this point in time, the configuration in the Alexa Developer Portal is about complete. The exception being the endpoint which doesn't exist yet.
+
+## Building a Lambda Function with Golang and MongoDB
+
+Alexa, for the most part should be able to direct requests, so now we need to create our backend to receive and process them. This is where Lambda, Go, and MongoDB come into play.
+
+Assuming Golang has been properly installed and configured, create a new project within your **$GOPATH** and within that project, create a **main.go** file. As boilerplate to get the ball rolling, this file should contain the following:
+
+```go
+package main
+
+func main() { }
+```
+
+With the boilerplate code added, now we can install the MongoDB Go driver. To do this, you could in theory do a `go get`, but the preferred approach as of now is to use the [dep](https://github.com/golang/dep) package management tool for Golang. To do this, after having installed the tool, execute the following:
+
+```bash
+dep init
+dep ensure -add "go.mongodb.org/mongo-driver/mongo"
+```
+
+We're using `dep` so that way the version of the driver that we're using in our project is version locked.
+
+In addition to the MongoDB Go driver, we're also going to need to get the [AWS Lambda SDK for Go](https://github.com/aws/aws-lambda-go) as well as an [unofficial SDK for Alexa](https://github.com/arienmalec/alexa-go), since no official SDK exists. To do this, we can execute:
+
+```bash
+dep ensure -add "github.com/arienmalec/alexa-go"
+dep ensure -add "github.com/aws/aws-lambda-go/lambda"
+```
+
+With the dependencies available to us, we can modify the project's **main.go** file. Open the file and add the following code:
+
+```go
+package main
+
+import (
+	"context"
+	"log"
+	"os"
+	"time"
+
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type Connection struct {
+	database *mongo.Database
+}
+
+func main() {
+	client, err := mongo.NewClient(options.Client().ApplyURI(os.Getenv("ATLAS_URI")))
+	if err != nil {
+		log.Fatal(err)
+	}
+	ctx, _ := context.WithTimeout(context.Background(), 10*time.Second)
```

Comment:
We're actually trying to deprecate the `Context` parameter for the `Connect` function because function doesn't do any blocking I/O, so the timeout is meaningless. Change to `context.Background()` or `context.TODO()` and remove the text below about the 10 second timeout.

Reply:
The documentation is still suggesting the use of `Context` for connecting:

https://github.com/mongodb/mongo-go-driver#usage

Can we have this updated for consistency?

Reply:
Yeah, that should be updated. We'll update it when we do the ticket to deprecate the parameter.